### PR TITLE
remove local reference to passing bundle and replace with working operator

### PR DESCRIPTION
**Pull request by komish**
The E2E tests previously used a "known good bundle", and the bundle Containerfile were stored in-repo. This is mostly fine, but it didn't make reference to a working operator with published assets.

This PR replaces that bundle reference with one that will need to be maintained https://github.com/opdev/simple-demo-operator. This will also run, and therefore should pass the DeployableByOLM test.

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -14,8 +14,8 @@ import (
 var _ = Describe("policy validation", func() {
 	Describe("When enforcing operator policy", func() {
 		var (
-			// TODO: replace this with a project-accessible namespace once identified.
-			goodImage = "quay.io/komish/preflight-test-bundle-passes:latest"
+			// TODO: replace the failure case with crafted examples.
+			goodImage = "quay.io/opdev/simple-demo-operator-bundle:v0.0.2"
```
Comment (samira-barouti) [inline test/e2e/e2e_test.go:18]: Out of curiosity: do we plan to have a pipeline or a manual procedure to create/update this bundle image?

**Review by samira-barouti — APPROVED** [review]
A small question

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -59,7 +59,8 @@ var _ = Describe("policy validation", func() {
 	Describe("When enforcing container policy", func() {
 
 		var (
-			// TODO: replace this with a project-accessible namespace once identified.
+			// TODO: replace the passing case with the container used by
+			// https://github.com/opdev/simple-demo-operator
 			goodImage = "quay.io/komish/preflight-test-container-passes:latest"
```
Comment (bcrochet) [inline test/e2e/e2e_test.go:64]: Do we want these to reference opdev?

**Review by sebrandon1 — APPROVED** [review]

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (komish) [reply test/e2e/e2e_test.go:64]: No, they don't exist in opdev yet =\

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**